### PR TITLE
feat: use `make install-slim` to speed up (ci) build

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

This PR adds a `slim` feature that strips out significant dependencies to minimize compile time for CI and local testing

Changes introduced in this pull request:

- Implement `slim` cargo feature and `make install-slim` (and make a few dependencies optional)
- Optimize `rustc` options for the `build-ubuntu` CI step to reducing linking time (from ~4min to ~2min in CI, getting accurate numbers requires `-Z time-passes` flag with the nightly toolchain) 

(when sccache build cache is mostly hit)
time cost in `build-ubuntu` step (on the critical path) is reduced from ~9min to ~5min (CI logs [before](https://github.com/ChainSafe/forest/actions/runs/8710839280/job/23893607330) [after](https://github.com/ChainSafe/forest/actions/runs/8714733437/job/23905459111?pr=4218))

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/cli/main.rs [hunk 1]
```diff
@@ -44,6 +44,7 @@ where
                 Subcommand::Send(cmd) => cmd.run(rpc::Client::from(api)).await,
                 Subcommand::Info(cmd) => cmd.run(api).await,
                 Subcommand::Snapshot(cmd) => cmd.run(api).await,
+                #[cfg(feature = "attach")]
                 Subcommand::Attach(cmd) => cmd.run(api),
```
Comment (lemmih) [inline src/cli/main.rs:48]: Let's keep the command and print a user-friendly error message if it is used when compiled without the `attach` feature flag.

File: scripts/tests/calibnet_other_check.sh [hunk 2]
```diff
@@ -13,8 +13,8 @@ forest_init
 echo "Verifying the non calibnet snapshot (./test-snapshots/chain4.car) is being served properly."
 $FOREST_CLI_PATH chain read-obj -c bafy2bzacedjrqan2fwfvhfopi64yickki7miiksecglpeiavf7xueytnzevlu
 
-echo "Testing js console"
-$FOREST_CLI_PATH attach --exec 'showPeers()'
+# echo "Testing js console"
+# $FOREST_CLI_PATH attach --exec 'showPeers()'
```
Comment (lemmih) [inline scripts/tests/calibnet_other_check.sh:17]: Hmm, it is problematic to have features that we are not testing. If we stop testing the JS integration, we should delete it from Forest.

File: src/cli_shared/cli/mod.rs [hunk 3]
```diff
@@ -125,12 +125,15 @@ pub struct CliOpts {
...
     #[arg(long)]
     pub tokio_console: bool,
     /// Send telemetry to `grafana loki`
+    #[cfg(feature = "tracing-loki")]
     #[arg(long)]
     pub loki: bool,
     /// Endpoint of `grafana loki`
+    #[cfg(feature = "tracing-loki")]
     #[arg(long, default_value = "http://127.0.0.1:3100")]
     pub loki_endpoint: String,
```
Comment (lemmih) [inline src/cli_shared/cli/mod.rs:138]: The user experience of conditional command-line flags is poor. The flags should be static and sensible errors should be printed if, say, the loki flag is used with enabling `tracing-loki`.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: src/cli_shared/cli/mod.rs (hunk 3 above)
Comment (hanabi1224) [reply src/cli_shared/cli/mod.rs]: Fixed.

File: scripts/tests/calibnet_other_check.sh (hunk 2 above)
Comment (hanabi1224) [reply scripts/tests/calibnet_other_check.sh]: Added this back

File: src/cli/main.rs (hunk 1 above)
Comment (hanabi1224) [reply src/cli/main.rs]: Fixed.

**Review by sudo-shashank — APPROVED** [review]

**Review by lemmih — APPROVED** [review]

**Pull request merged**